### ISSUE TITLE
Upgrading from 10.0.7 to 10.0.9 : SQL Error "1264"

### ISSUE THREAD

**Issue by Krafting** [glpi-project/glpi#15202]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.9

### Bug description

![image](https://github.com/glpi-project/glpi/assets/36538123/75aa2a71-f7e7-4897-b5cc-c522a69c8f71)

I get this error when trying to update from 10.0.7 to 10.0.9:

```
Update to 10.0.8

Change of the database layout - glpi_computervirtualmachines (0 seconds)
SQL Error "1264": Out of range value for column 'ram' at row 2 in query "ALTER TABLE `glpi_computervirtualmachines` CHANGE `ram` `ram` int unsigned DEFAULT NULL "
10.0.8 multiple alter in glpi_computervirtualmachines - Error during the database query: ALTER TABLE `glpi_computervirtualmachines` CHANGE `ram` `ram` int unsigned DEFAULT NULL - Error is Out of range value for column 'ram' at row 2 
```

It also happened when updating from 10.0.7 to 10.0.8.

If you need more informations, please ask me, i'll be glad to privide it.

Maybe related (similar issue with the RAM in database): https://github.com/glpi-project/glpi/pull/14975

**Comment by cedric-anne** [comment]
Hi,

Can you give us the result of the following query: `` SELECT DISTINCT `ram`, COUNT(`id`) FROM `glpi_computervirtualmachines` WHERE `ram` REGEXP '[^0-9]+' GROUP BY `ram`; ``?

**Comment by Krafting** [comment]
Hi,

Here is the output, empty:

![image](https://github.com/glpi-project/glpi/assets/36538123/7dbbe3ae-3809-42bb-8a9f-5488a3439023)

**Comment by Krafting** [comment]
Here is the same request without the WHERE clause, if that helps:

![image](https://github.com/glpi-project/glpi/assets/36538123/306e9613-a905-418e-b063-87ff01042468)

**Comment by Krafting** [comment]
These number seems to be from all the LXC on my main server running Proxmox. If that helps.

Here is a sample from the server Virtualization page:

![image](https://github.com/glpi-project/glpi/assets/36538123/a881fe07-4a68-4937-86da-4e1be84964b1)

**Comment by cedric-anne** [comment]
What was the software used to inventory these virtual machines?
What it FusionInventory Agent, GLPI Inventory Agent, OCS, ...?

**Comment by Krafting** [comment]
Glpi-agent 1.4, it gets inventoried every 24 hours.

**Comment by cedric-anne** [comment]
Can you provide the inventory file for one of these virtual machines?

**Comment by Krafting** [comment]
Sure, however, how can I  do that?

**Comment by cedric-anne** [comment]
> Sure, however, how can I do that?

At the bottom of the corresponding computer main form, there is a button to download the corresponding inventory file:
![image](https://github.com/glpi-project/glpi/assets/33253653/7f553763-818d-43d6-90a5-464234bc9d94)

**Comment by Krafting** [comment]
Thanks, there you go!

It was done using GLPI-Agent 1.5, I've just updated it.

[computer_0_2.json.log](https://github.com/glpi-project/glpi/files/12036225/computer_0_2.json.log)

GitHub doesn,'t let me upload .json file, I've added .log at the end to be able to upload it.

Note: I've tried to delete a LXC in the Virtualizatrion Tab and then re-inventoried, and the LXC came back with no RAM at all. (Tried with a Running LXC and a stopped one.)

**Comment by trasher** [comment]
There is no memory for any of your containers in provided inventory file.

**Comment by trasher** [comment]
Please try #15211

**Comment by g-bougard** [comment]
Hi @Krafting 

the too high values seems to come from old inventories run by FusionInventory agent or older GLPI-Agent version. I fixed the LXC container memory reading in GLPI-Agent 1.3 after that issue glpi-project/glpi-agent#132.

**Comment by cedric-anne** [comment]
#15211 should fix the migration, by removing invalid values.

It seems that memory value is not anymore provided by inventory agent. I have no idea if it is a normal behaviour. When this value is missing from inventory file, maybe it should also be removed in GLPI, but I am not sure of that.

**Comment by Krafting** [comment]
Hello,

How can I try #15211 ? 
Just.by downloading 10.0.9 and replacing the edited file ?

I'll try this later today and get back to you!


Thanks

**Comment by cedric-anne** [comment]
You can indeed, in this specific case, replace the whole file, or you can download a patch file for proposed change (https://github.com/glpi-project/glpi/pull/15211.patch) and apply it using the `patch` utility.

**Comment by cedric-anne** [comment]
Migration issue is fixed by #15211 .

Removal of invalid values during inventory process will be fixed later.
